### ISSUE TITLE
How to use more than 1 worker in Jupyter?

### ISSUE THREAD

**Issue by christophechu** [dylkot/cNMF#90]

**Comment by wudustan** [comment]
I want to +1

I followed along with the Jupyter example for PBMC data. I used the command:

`cnmf_obj.factorize(worker_i=0, total_workers=16)`

And the output I got was along these lines:

```
[Worker 0]. Starting task 0.
[Worker 0]. Starting task 16.
[Worker 0]. Starting task 32.
[Worker 0]. Starting task 48.
[Worker 0]. Starting task 64.
[Worker 0]. Starting task 80.
[Worker 0]. Starting task 96.
...
```
Is it fair to assume that this means the function is being ran over 16 cores? I'm not necessarily seeing a bit spike in core utilisation on my machine.

**Comment by livewei** [comment]
I want to +1

**Comment by chongyiyang1117** [comment]
use another func in cnmf.py 
------>cnmf_obj.factorize_multi_process(total_workers=4)

**Comment by wudustan** [comment]
Is there any update avaialable for this?

**Comment by liezeltamon** [comment]
+ 1

**Comment by liezeltamon** [comment]
> I want to +1
> 
> I followed along with the Jupyter example for PBMC data. I used the command:
> 
> `cnmf_obj.factorize(worker_i=0, total_workers=16)`
> 
> And the output I got was along these lines:
> 
> ```
> [Worker 0]. Starting task 0.
> [Worker 0]. Starting task 16.
> [Worker 0]. Starting task 32.
> [Worker 0]. Starting task 48.
> [Worker 0]. Starting task 64.
> [Worker 0]. Starting task 80.
> [Worker 0]. Starting task 96.
> ...
> ```
> 
> Is it fair to assume that this means the function is being ran over 16 cores? I'm not necessarily seeing a bit spike in core utilisation on my machine.
> 
> [@dylkot](https://github.com/dylkot) would you be able to comment if you have a second? Or could you point me to any document that discusses the avaialble arguments and their use for the python library.

I don't think this generates all the iterations across all specified k's based on the output I get when I parallelise like this..

**Comment by wudustan** [comment]
@liezeltamon I agree I definitely think I am doing something wrong but I have posted my issue elsewhere and haven't had any luck in getting a response.

**Comment by jiahaoh** [comment]
after reading the source code, I believe it is `cnmf_obj.factorize_multi_process(total_workers=4)`
